### ISSUE TITLE
Images with query strings are given the wrong extension

### ISSUE THREAD

**Issue by alecgibson** [cyrilis/epub-gen#53]
When `epub-gen` is provided an `img` with a `src` with a query string, then the image's extension is incorrectly changed to `.bin`.

For example:

```html
<img src="http://example.com/image.jpg?w=400">
```

I believe the bit of the code to blame is [here](https://github.com/cyrilis/epub-gen/blob/128afec314c31b14569b07376adeb179a0e8887f/lib/index.coffee#L145-L146), where the [`mime`](https://github.com/broofa/node-mime) package returns:

```javascript
mime.lookup("http://example.com/image.jpg")       // => "image/jpeg"
mime.lookup("http://example.com/image.jpg?w=400") // => "application/octet-stream"

mime.extension("image/jpeg")                      // => "jpeg"
mime.extension("application/octet-stream")        // => "bin"
```

We can try to strip the query string at our end, but sometimes query strings are used to contain options for how the image is rendered (a common example being [Gravatar](https://en.gravatar.com/)).

It would be best if `epub-gen` stripped the query string when trying to determine the `mediaType` and `extension`.

**Comment by pedrosanta** [comment]
Nice catch @alecgibson. 👍 Yep, I say strip the query string from the image URL, and then `mime` would work well, I suppose. I can put up a PR, along with another one for #52, as soon as possible.

**Comment by pedrosanta** [comment]
> It would be best if `epub-gen` stripped the query string when trying to determine the `mediaType` and `extension`.

Yep, just checked out on [`mime`](https://github.com/broofa/node-mime) package and it really doesn't support URL with query strings (check broofa/node-mime#131)(1), so, agreed, better to update the lib to remove/ignore the query string before checking the mime types/extension.

Coming right up!

(1) - The `mime` lib seems more oriented to file paths and all, and author prefers that to lie outside the `mime` lib so, no point raising an issue there/upstream, I suppose. 🙂

**Comment by pedrosanta** [comment]
Fixed on [v0.0.19](https://github.com/cyrilis/epub-gen/releases/tag/v0.0.19). Closing.
